zkevm-circuits/src/evm_circuit/step.rs

```diff
@@ -482,14 +482,15 @@ pub(crate) struct StepRow<F> {
 pub(crate) struct Step<F> {
     pub(crate) state: StepState<F>,
     pub(crate) rows: Vec<StepRow<F>>,
+    pub(crate) rotation_offset: usize,
```

Comment:
We compute it to add it to the height count of the Step? Is my understanding correct?

Reply:
Correct, because the state variables are set here the constraint builder doesn't directly know the full height of the step. The total height of a step is the number of rows used for the state variables + the number of rows used by the constraint builder.

This is just a temporary solution for this PR to be honest, in https://github.com/appliedzkp/zkevm-circuits/pull/425 where everything comes together all cell management is done in a dedicated `CellManager` which manages all cells in a step and so the height calculation can be handled more cleanly.